### PR TITLE
Fix doxygen warnings and invalid doxygen commands

### PR DESCRIPTION
**Pull request by 9prady9**

### CODE REVIEW

File: include/af/blas.h [hunk 1]
```diff
@@ -158,8 +158,6 @@ namespace af
         \note optLhs = AF_MAT_CONJ and optRhs = AF_MAT_NONE will run conjugate dot operation.
         \note This function is not supported in GFOR
 
-        \returns out = dot(lhs, rhs)
```
Comment (umar456) [inline include/af/blas.h]: It would be nice to replace this with a sentence

\returns dot product of lhs and rhs

File: include/af/blas.h [hunk 2]
```diff
@@ -190,8 +188,6 @@ namespace af
...
```
Comment (umar456) [inline include/af/blas.h]: Same as above

File: include/af/data.h [hunk 3]
```diff
@@ -415,18 +415,10 @@ namespace af
...
     AFAPI void replace(array &a, const array  &cond, const double &b);
 #endif
-
-    /**
-      @}
-    */
 }
-#endif
 
-#ifdef __cplusplus
```
Comment (umar456) [inline include/af/data.h]: This ifdef is necessary isn't it?

File: include/af/data.h [hunk 4]
```diff
@@ -415,18 +415,10 @@ namespace af
     */
     AFAPI void replace(array &a, const array  &cond, const double &b);
 #endif
-
-    /**
-      @}
-    */
 }
-#endif
```
Comment (umar456) [inline include/af/data.h]: I think this is also necessary. It excludes the af namespace from C headers.

File: include/af/blas.h (hunk 1 above)
Comment (9prady9) [reply include/af/blas.h]: There is a sentence already, what I removed is duplicate statement

File: include/af/data.h (hunk 4 above)
Comment (9prady9) [reply include/af/data.h]: No, its not needed, but we can keep them for readability.

**Pull request merged**